CMakeLists.txt

```diff
@@ -255,6 +255,7 @@ if(USE_MKLDNN)
   add_subdirectory(3rdparty/mkldnn)
 
   include_directories(3rdparty/mkldnn/include)
+  include_directories(${PROJECT_BINARY_DIR}/3rdparty/mkldnn/include)
```

Comment:
In this folder, there is a new header file of MKL-DNN which is automatically generated during compilation. I doubt it would be used by Horovod and should be distributed with pip package. @szha @yuxihu